### ISSUE TITLE
Can't compile DocuementServer - unbuntu 22.04

### ISSUE THREAD

**Issue by agusdomin** [ONLYOFFICE/build_tools#567]
Describe your problem:
When i ran for second time `./automate.py server` 
![image](https://user-images.githubusercontent.com/50929799/202453761-b6c728a5-4e8b-4e57-86a9-76ca45cf5f02.png)

I ran for second time automate.py script because it did not create a " ./out " directory anywhere.

First time i tried to compile the server, i reproduced every step like it says here:
https://helpcenter.onlyoffice.com/installation/docs-community-compile.aspx?utm_source=blog&utm_medium=social&utm_campaign=build_tools

build_tools base repo version: latest

Operating system: Ubuntu 22.04 on Docker container

**Comment by ShockwaveNN** [comment]
I think it may be the same problem as https://github.com/ONLYOFFICE/build_tools/issues/550

Not enought RAM, Swap, or disk space

Especially if this failed on v8 compliation

Also we use Ubuntu 16.04 as our base of compilation, so better to check in on it
We got Dockerfile in this repo and instruction in Readme how to do so

**Comment by agusdomin** [comment]
> I think it may be the same problem as #550
> 
> Not enought RAM, Swap, or disk space
> 
> Especially if this failed on v8 compliation
> 
> Also we use Ubuntu 16.04 as our base of compilation, so better to check in on it We got Dockerfile in this repo and instruction in Readme how to do so

Thanks, right now i'm compiling on Ubuntu 16, another question

If a want to rewrite some code and compile changes, i will have to run automate.py again, am not?

**Comment by ShockwaveNN** [comment]
@agusdomin 

> If a want to rewrite some code and compile changes, i will have to run automate.py again, am not?

Yeah, but if I remember correctly it will compile only change parts, but still it can take time, if it's in c++ `core` subproject, but it's fast if it's just a javascript changes in `sdkjs` or `web-apps`

**Comment by agusdomin** [comment]
Thanks a lot, i will confirm you if i can compile it successfully

**Comment by ChiuJun** [comment]
@agusdomin I think you can check the source code directory that the script is compiling with.(OpenSSL? maybe,I'm not sure)
I had a similar situation before, because of the dirty directory which is due to bad network or compilation interrupted.

**Comment by agusdomin** [comment]
i had a error starting Docservice

![imagen](https://user-images.githubusercontent.com/50929799/202768661-1cf4171a-47e2-43fb-96eb-ccc52134e7ba.png)

I thing it because i changed a secret in default.json, isn't it?
services.CoAuthoring.secret.inbox: {"string": "secret", "file": ""},
*.outbox: {"string": "secret", "file": ""},
*.session: {"string": "secret", "file": ""}

**Comment by ShockwaveNN** [comment]
The error looks like you don't start a PostgreSQL, 5432 is psql default port

**Comment by agusdomin** [comment]
Yes it was that thx, and sorry but now when a i try to connect nextcloud to doc server, throws me this error:
" error while downloading the document file to be converted "

And DocService proccess gives me this logs:

![imagen](https://user-images.githubusercontent.com/50929799/203340126-3b8148b0-2be8-4b3c-916f-205426c73cad.png)

**Comment by ShockwaveNN** [comment]
I think this error is not problem on DOcumentServer side, but DocumentServer for some reason cannot access nextcloud server and file by it's url, you can try to download it via `wget` and I think some error will arise, like 403

**Comment by agusdomin** [comment]
Yeah that right, from DocServer container it's give me a 403 error. Something strange it's that code and connect undefined

![imagen](https://user-images.githubusercontent.com/50929799/203341897-8a3dd9aa-c8d7-4638-914c-5b171b686d06.png)

**Comment by ShockwaveNN** [comment]
Remove everything after :attempt-3 including it and recheck

**Comment by ShockwaveNN** [comment]
Yep if your `wget` can't download file it's defenetly not a problem in DocumentServer but in your networking

**Comment by agusdomin** [comment]
Sorry but what file it's trying to download?

**Comment by ShockwaveNN** [comment]
The file you've trying to edit

**Comment by agusdomin** [comment]
but i'm no trying to edit a file, i'm trying to integrate DocServer to Nextcloud server. Like this.
![imagen](https://user-images.githubusercontent.com/50929799/203355203-d77debf9-722e-4418-af13-46764c5917f5.png)

**Comment by agusdomin** [comment]
It was a problem with tokens and secret between Nextcloud server and Onlyoffice Documet server. 

Something that help me:
https://forum.onlyoffice.com/t/unable-to-use-onlyoffice-with-nextcloud/2089/9

if you want you can close this issue so it not will become something big and diverse. If i have another problem i will create a new one

**Comment by ShockwaveNN** [comment]
Yeah, I'll move it to discussion since there is not any specific issue, but more like Q&A
